Alex

n00b001 opened the pull request.

smallpaul committed:
ball history shiz

Commit:
implement score deducting with game ticks
implement self play against random individuals from the hall of fame (updated at the end of the generations)

smallpaul committed:
ball history and remove lots of wanted stuff. oops

Commit:
Merge remote-tracking branch 'origin/ball-history' into alex

# Conflicts:
#	config.py
#	main.py

Commit:
merged paul's changes - now use previous ball location as input

smallpaul committed:
load built in AI, hard coded AI and famous individuals

Commit:
push me

smallpaul committed:
Merge remote-tracking branch 'origin/alex' into ball-history

smallpaul committed:
IGNORE

smallpaul committed:
* save and load hall of fame.
* remove open cv
* new fitness equation

smallpaul committed:
start uring rint again
don't scale screen
don't use opencv
left player correclty has ball pos

Commit:
push me

Commit:
best one

Commit:
push me

Commit:
fix FATAL BUG

smallpaul committed:
Merge remote-tracking branch 'origin/alex' into ball-history

smallpaul committed:
automaticepsilonlexicase implimented, easy my dude
crossover blend - done
biases, sorted.

smallpaul committed:
automaticepsilonlexicase implimented, easy my dude
crossover blend - done
biases, sorted.

Commit:
use argmax

smallpaul committed:
automaticepsilonlexicase implimented, easy my dude
crossover blend - done
biases, sorted.

Commit:
reward is related to how much score the enemy has previously got

Commit:
fix bug related to loading random hall of famer

Commit:
saving/loading network shape
renamed some variables/constants
unvectorize sigmoid (there is a non-breaking bug where the input gives an error because the output can't be represented into a double)

Commit:
there was a bug in selecting an opponent from the hall_of_fame

Commit:
checkpoints are put in their own folder (so the base fodler can be excluded)
big refactor
pickle inspector can load best now
and there was a bug loading NN shape and calculating gene amount, which is fixed
there was another bug when they drew, but that's fixed too
I've made it scale the rendered image
the FPS is takes into consideration the calculation time

Commit:
update gitignore

Commit:
compatible with <python36

Commit:
fix bug where left player doesn't see ball correctly (also bug impacted right player)
the object reference was copied and modification was performed on that.  Rather than creating a new object

Commit:
actaully fixed the bug where left player gets confused
also fixed left-model not loaded when there is no hall_of_fame models to load

smallpaul reviewed (approved):
tl:dr 
looks good to me

smallpaul committed:
trying to fix first run issues

smallpaul committed:
fixed first run issue i think

smallpaul committed:
fixed first run issue i think

smallpaul committed:
Merge remote-tracking branch 'origin/alex' into paul

Commit:
allow to be run on server
I got confused and tried to Y flip

Commit:
checkpoints -> checkpoints/checkpoints

smallpaul committed:
hopefully fixes but i imagine ive broken alot too

Commit:
use scoop logging

Commit:
import scoop

Commit:
fix printing

Commit:
fix printing

smallpaul committed:
human play

smallpaul committed:
add pynput to requirements

smallpaul committed:
undoing my changes to hall of fame

smallpaul committed:
human play fixes

Commit:
Merge remote-tracking branch 'origin/paul' into alex

Commit:
refactor, clean, remove sleep

Commit:
setup to sidebyside test get_rect

Commit:
improve perf

Commit:
done with testing framework

Commit:
made testing generic

Commit:
change to sigmoid
loading checkpoints was wrong path
change ga params
more logs!

Commit:
aggressively delete unused variables

smallpaul commented on ga.py:
```diff
@@ -0,0 +1,99 @@
+        return None
+
+
+toolbox = base.Toolbox()
```
is there down sides to declaring things like this? outside of functions i mean

n00b001 replied:
This is a global variable
